#! /usr/bin/env python

PACKAGE = 'point_cloud_filter'

import os
import sys
sys.path.insert(0, os.path.dirname(__file__))
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator ()

gen.add("enabled",  bool_t,     0, "Whether to apply this filter or not",  True)
gen.add("min_x",    double_t,   0, "Body filter min x",     -0.6)
gen.add("max_x",    double_t,   0, "Body filter max x",      0.25)
gen.add("min_y",    double_t,   0, "Body filter min y",     -0.3)
gen.add("max_y",    double_t,   0, "Body filter max y",      0.5)
gen.add("min_z",    double_t,   0, "Body filter min z",     -0.3)
gen.add("max_z",    double_t,   0, "Body filter max z",      0.0)
gen.add("rotation", double_t,   0, "Body filter rotation",  -0.785398)

exit (gen.generate (PACKAGE, "pcl_ros", "BodyFilter"))